acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/privacy/multitenancy/OnChainMultiTenancyAcceptanceTest.java

```diff
@@ -0,0 +1,697 @@
+import org.web3j.protocol.core.methods.response.EthCall;
+import org.web3j.utils.Base64String;
+
+public class OnChainMultiTenancyAcceptanceTest extends OnChainPrivacyAcceptanceTestBase {
+
+  private static final String eventEmitterDeployed =
+      "0x6080604052600436106100565763ffffffff7c01000000000000000000000000000000000000000000000000000000006000350416633fa4f245811461005b5780636057361d1461008257806367e404ce146100ae575b600080fd5b34801561006757600080fd5b506100706100ec565b60408051918252519081900360200190f35b34801561008e57600080fd5b506100ac600480360360208110156100a557600080fd5b50356100f2565b005b3480156100ba57600080fd5b506100c3610151565b6040805173ffffffffffffffffffffffffffffffffffffffff9092168252519081900360200190f35b60025490565b604080513381526020810183905281517fc9db20adedc6cf2b5d25252b101ab03e124902a73fcb12b753f3d1aaa2d8f9f5929181900390910190a16002556001805473ffffffffffffffffffffffffffffffffffffffff191633179055565b60015473ffffffffffffffffffffffffffffffffffffffff169056fea165627a7a72305820c7f729cb24e05c221f5aa913700793994656f233fe2ce3b9fd9a505ea17e8d8a0029";
+
+  private static final PermissioningTransactions permissioningTransactions =
+      new PermissioningTransactions();
+  private static final long VALUE_SET = 10L;
+
+  private PrivacyNode alice;
+  private MultiTenancyPrivacyNode aliceMultiTenancyPrivacyNode;
+
+  @Before
+  public void setUp() throws Exception {
+    alice =
+        privacyBesu.createOnChainPrivacyGroupEnabledMinerNode(
+            "node1", PrivacyAccountResolver.MULTI_TENANCY, Address.PRIVACY, true);
+    final BesuNode aliceBesu = alice.getBesu();
+    privacyCluster.startNodes(alice);
+    final String alice1Token =
+        aliceBesu.execute(permissioningTransactions.createSuccessfulLogin("user", "pegasys"));
+    aliceBesu.useAuthenticationTokenInHeaderForJsonRpc(alice1Token);
+    final String alice2Token =
+        aliceBesu.execute(permissioningTransactions.createSuccessfulLogin("user2", "Password2"));
+    final String alice3Token =
+        aliceBesu.execute(permissioningTransactions.createSuccessfulLogin("user3", "Password3"));
+    privacyCluster.awaitPeerCount(alice);
+
+    final String alice1EnclaveKey = alice.getOrion().getPublicKeys().get(0);
+    final String alice2EnclaveKey = alice.getOrion().getPublicKeys().get(1);
+    final String alice3EnclaveKey = alice.getOrion().getPublicKeys().get(2);
+
+    aliceMultiTenancyPrivacyNode = new MultiTenancyPrivacyNode(alice);
+    aliceMultiTenancyPrivacyNode
+        .addTenantWithToken(alice1EnclaveKey, alice1Token)
+        .addTenantWithToken(alice2EnclaveKey, alice2Token)
+        .addTenantWithToken(alice3EnclaveKey, alice3Token);
+  }
+
+  @After
+  public void tearDown() {
+    privacyCluster.close();
+  }
+
+  @Test
+  public void createPrivacyGroup() {
+    createOnChainPrivacyGroup(alice);
+  }
+
+  @Test
+  public void createPrivacyWithAllTenants() {
+    final MultiTenancyPrivacyGroup justAlice = new MultiTenancyPrivacyGroup();
```

Comment:
`justAlice` doesn't seem like the right name. Maybe `privacyGroup` ?

Reply:
Done